packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1468,6 +1478,38 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
     return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
   }
 
+  SelectionResult _updateSelectionEdgeByWord(Offset globalPosition, {required bool isEnd}) {
+    _setSelectionPosition(null, isEnd: isEnd);
+    final Matrix4 transform = paragraph.getTransformTo(null);
+    transform.invert();
+    final Offset localPosition = MatrixUtils.transformPoint(transform, globalPosition);
+    if (_rect.isEmpty) {
+      return SelectionUtils.getResultBasedOnRect(_rect, localPosition);
+    }
+    final Offset adjustedOffset = SelectionUtils.adjustDragOffset(
+      _rect,
+      localPosition,
+      direction: paragraph.textDirection,
+    );
+    final TextPosition position = _clampTextPosition(paragraph.getPositionForOffset(adjustedOffset));
+
+    final (TextPosition start, TextPosition end) wordBoundary = _getWordBoundaryAtPosition(position);
+    TextPosition targetPosition = isEnd ? wordBoundary.$2 : wordBoundary.$1;
+    if (!_rect.contains(localPosition)) {
```

Comment:
I am not entirely sure what this is for can you explain?

Reply:
I found that since `SelectionUtils.adjustDragOffset` returns the top left of the rect for `ltr` if the drag is in `Area1` of a `Selectable` it will return `TextPosition(0)`. When using `_getWordBoundaryAtPosition(position)` that ends up selecting the first word of the `selectable`. 

This is not desired since say we have a list of `Selectables` and our drag starts at the first one, dragging down to the second one moves the selection to the word on the second selectable, but then when dragging back up to the first selectable the second selectables first word is still selected because `_getWordBoundaryAtPosition(TextPosition(0))` = `first word`.

This same logic is in `_updateSelectionEdge` but it works there because the edge will moved to `TextPosition(0)` and not the word at `TextPosition(0)`.

This was my workaround for this issue.

Reply:
Thanks for explanation. If this is the case, I think you can move this before the whole `_getWordBoundaryAtPosition` and short circuit it. Also would be good to add some comments about this